sbpy/units/tests/test_core.py

```diff
@@ -96,3 +97,83 @@ def test_spectral_density_vega_bp(filename, fluxd, to, tol):
+        1.86599755e-12 * u.Unit('W/(m2 Hz)'), None, 0.02809415),
+    (3.4 * VEGAmag, None, None, -26.77471503 * VEGAmag, 0.02865984),
+    (3.4 * u.ABmag, None, None, -26.77471503 * u.ABmag, 0.02865984),
+))
+def test_reflectance_ref(mag, wfb, f_sun, M_sun, ref):
+    """Test conversion from flux to reflectance
+
+    Use Ceres as the reference: H = 3.4 mag, radius = 460 km, average
+    bidirectional reflectance at zero phase angle 0.029 (~1/pi of geometric
+    albedo).
+    """
+    xsec = 6.648e5 * u.km**2
+    r = mag.to('1/sr', reflectance(cross_section=xsec, wfb=wfb, f_sun=f_sun,
+        M_sun=M_sun))
+    assert r.unit == u.sr**-1
+    assert np.isclose(r.value, ref)
+
+
+@pytest.mark.parametrize('mag, wfb, f_sun, M_sun, radius', (
+    (3.4 * VEGAmag, None, None, None, 460.01351274),
+    (3.4 * VEGAmag, 'johnson_v', None, None, 460.01351274),
+    (3.4 * VEGAmag, 5500 * u.AA, None, None, 452.62198065),
+    (1.56644783e-09 * u.Unit('W/(m2 um)'), None,
+        1839.93273227 * u.Unit('W/(m2 um)'), None, 460.01351274),
+    (1.55728147e-24 * u.Unit('W/(m2 Hz)'), None,
+        1.86599755e-12 * u.Unit('W/(m2 Hz)'), None, 455.45095634),
+    (3.4 * VEGAmag, None, None, -26.77471503 * VEGAmag, 460.01351274),
+    (3.4 * u.ABmag, None, None, -26.77471503 * u.ABmag, 460.01351274),
+))
+def test_reflectance_xsec(mag, wfb, f_sun, M_sun, radius):
+    """Test conversion from flux to reflectance
+
+    Use Ceres as the reference: H = 3.4 mag, radius = 460 km, average
+    bidirectional reflectance at zero phase angle 0.029 (~1/pi of geometric
+    albedo 0.09).
+    """
+    ref = 0.02865984 / u.sr
+    xs = mag.to('km2', reflectance(reflectance=ref, wfb=wfb, M_sun=M_sun))
+    ra = np.sqrt(xs/np.pi)
+    assert ra.unit == u.km
+    assert np.isclose(ra.value, radius)
+
+
+def test_reflectance_spec():
+    """Test conversion from flux spectrum to reflectance spectrum
+
+    Use Tempel 1 spectrum collected by Deep Impact HRI-IR at
+    2005-07-04 05:32 UT as the standard.  Data file
+    'data/hi05070405_9000036-avg-spec.txt'
+    """
+    fn = get_pkg_data_filename(os.path.join(
+        'data', 'hi05070405_9000036-avg-spec.txt'))
+    t1 = ascii.read(fn)
+    ifov = 1e-5 * u.rad
+    delta = 15828 * u.km
+    rh = 1.5 * u.au
+    wave = t1['wave'] * u.um
+    spec = (t1['spec'] * u.Unit('W/(m2 um sr)') * ifov**2).to('W/(m2 um)',
+        u.dimensionless_angles()) * delta.to('au').value**2 \
+        * rh.to('au').value**2
+    spec_nu = spec.to('W/(m2 Hz)', u.spectral_density(wave))
+    xsec = (ifov * delta).to('km', u.dimensionless_angles())**2
+    ref1 = spec.to('1/sr', reflectance(cross_section=xsec, wfb=wave))
+    ref2 = spec.to('1/sr', reflectance(cross_section=xsec, f_sun=t1['flux_sun']*u.Unit('W/(m2 um)')))
+    ref3 = spec_nu.to('1/sr', reflectance(cross_section=xsec, f_sun=t1['flux_sun_nu']*u.Unit('W/(m2 Hz)')))
+    assert ref1.unit == '1/sr'
+    assert np.isclose(ref1.value, t1['ref']).all()
```

Comment:
FYI, there is also an `np.allclose()` function.

Reply:
Revised for `np.allclose()`.  Thanks for pointing it out to me.